lib/hooks/deleteHighlight.ts

```diff
@@ -0,0 +1,26 @@
+import { supabase } from "lib/utils/supabase";
+
+const baseUrl = process.env.NEXT_PUBLIC_API_URL;
+const deleteHighlight = async (id: string) => {
+  const sessionResponse = await supabase.auth.getSession();
+  const sessionToken = sessionResponse?.data.session?.access_token;
+  try {
+    const res = await fetch(`${baseUrl}/user/highlights/${id}`, {
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${sessionToken}`
+      },
+      method: "DELETE"
+    });
+
+    if (res.ok) {
+      return true;
+    }
```

Comment:
if res is not ok, what happens? Nothing is returned. IMO should throw an error.

Reply:
We should catch the errors and display the error message.